src/event.rs

```diff
@@ -74,9 +74,105 @@ pub enum Event<'a, T: 'static> {
+    /// destroyed, which indirectly invalidates any existing render surfaces that may have been
+    /// created outside of Winit (such as an `EGLSurface`, [`VkSurfaceKHR`] or [`wgpu::Surface`]).
+    ///
+    /// After being `Suspended` on Android applications must drop all render surfaces before
+    /// the event callback completes, which may be re-created when the application is next [`Resumed`].
+    ///
+    /// [`SurfaceView`]: https://developer.android.com/reference/android/view/SurfaceView
+    /// [Activity lifecycle]: https://developer.android.com/guide/components/activities/activity-lifecycle
+    /// [`VkSurfaceKHR`]: https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkSurfaceKHR.html
+    /// [`wgpu::Surface`]: https://docs.rs/wgpu/latest/wgpu/struct.Surface.html
+    ///
+    /// ## iOS
+    ///
+    /// On iOS the `Suspended` event is currently emitted in response to an
+    /// [`applicationWillResignActive`] callback which means that the application is
+    /// about to transition from the active to inactive state (according to the
+    /// [iOS application lifecycle]).
+    ///
+    /// [`applicationWillResignActive`]: https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622950-applicationwillresignactive
+    /// [iOS application lifecycle]: https://developer.apple.com/documentation/uikit/app_and_environment/managing_your_app_s_life_cycle
+    ///
+    /// [`Resumed`]: Self::Resumed
     Suspended,
 
     /// Emitted when the application has been resumed.
+    ///
+    /// For consistency all platforms emit a `Resumed` event even if they don't themselves have a
+    /// formal suspend/resume lifecycle. For systems without a standard suspend/resume lifecycle
+    /// the `Resumed` event is is always emitted after the [`NewEvents(StartCause::Init)`][StartCause::Init]
+    /// event.
+    ///
+    /// # Portability
+    ///
+    /// It's recommended that applications should only initialize their graphics context and create
+    /// a window after they have received their first `Resumed` event. Some systems
+    /// (specifically Android) won't allow applications to create a render surface until they are
+    /// resumed.
+    ///
+    /// Considering that the implementation of [`Suspended`] and `Resumed` events may be internally
+    /// driven by multiple platform-specific events, and there may be subtle differences across
+    /// platforms with how these internal events are delivered, it's recommended that applications
+    /// be able to gracefully handle redundant (i.e. back-to-back) [`Suspended`] or `Resumed` events.
+    ///
+    /// Also see [`Suspended`] notes.
+    ///
+    /// ## Android
+    ///
+    /// On Android specifically, the [`Suspended`] event relates to two things:
+    /// 1. The application has resumed (according to the [Activity lifecycle])
+    /// 2. And the application has an associated [`SurfaceView`] (which is expected to closely correlate
```

Comment:
Perhaps my comment above on the similar docs for `Suspended` are simply resolved by having an `And` here as first word for `2.` (instead of `Or`), just like you have here.

Reply:
The choice of "Or" above and "And" was _very_ _very_ intentional :-) , those are the words that reflect the actual logic in each case.

I.e.:
- when being suspended you _might_ have had your surface view destroyed (one of two possible reasons to suspend - logical OR).
- when resuming you _definitely_ do have a surface view (there are two factors that must _both_ be satisfied before we resume - logical AND)

Reply:
Though actually one glaring issue here is that the first line is a copy and paste from the `Suspended` event docs. This should  start with "On Android specifically, the **[`Resumed`]** event relates to two things"

Reply:
okey, I've tried adding another line to really try and clarify the meaning here as well as fix the copy and paste error.

```diff
--- a/src/event.rs
+++ b/src/event.rs
@@ -97,6 +97,9 @@ pub enum Event<'a, T: 'static> {
     /// 2. Or the application has had its associated [`SurfaceView`] destroyed (which is expected
     /// to closely correlate with being paused)
     ///
+    /// _(I.e. **either** of these may lead to a `Suspended` event, but applications should act
+    /// as if both have happened.)_
+    ///
     /// Applications that need to run on Android should assume their [`SurfaceView`] has been
     /// destroyed, which indirectly invalidates any existing render surfaces that may have been
     /// created outside of Winit (such as an `EGLSurface`, [`VkSurfaceKHR`] or [`wgpu::Surface`]).
@@ -145,11 +148,13 @@ pub enum Event<'a, T: 'static> {
     ///
     /// ## Android
     ///
-    /// On Android specifically, the [`Suspended`] event relates to two things:
+    /// On Android specifically, the `Resumed` event relates to two things:
     /// 1. The application has resumed (according to the [Activity lifecycle])
     /// 2. And the application has an associated [`SurfaceView`] (which is expected to closely correlate
     /// with being resumed)
     ///
+    /// _(I.e. **both** of these must happen before the application gets a `Resumed` event.)_
+    ///
     /// Applications that need to run on Android must wait until they have been `Resumed`
     /// before they will be able to create a render surface (such as an `EGLSurface`,
     /// [`VkSurfaceKHR`] or [`wgpu::Surface`]) which depend on having a
```